nilearn/image/image.py

```diff
@@ -674,6 +674,7 @@ def threshold_img(img, threshold, mask_img=None):
     img_data = np.nan_to_num(img_data)
 
     if mask_img is not None:
+        mask_img = check_niimg(mask_img)
```

Comment:
Maybe a check_niimg_3d rather? So that if the user gives a 4D image we get a meaningful error message.